TheraLang.BLL/Services/PaymentHistoryService.cs

```diff
@@ -0,0 +1,120 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Common.Exceptions;
+using System.Threading.Tasks;
+using TheraLang.BLL.DataTransferObjects;
+using TheraLang.BLL.Interfaces;
+using TheraLang.DAL.Entities;
+using TheraLang.DAL.UnitOfWork;
+using Common;
+
+namespace TheraLang.BLL.Services
+{
+    public class PaymentHistoryService : IPaymentHistoryService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IUserManagementService _userManagementService;
+
+        public PaymentHistoryService(IUnitOfWork unitOfWork, IUserManagementService userManagementService)
+        {
+            _unitOfWork = unitOfWork;
+           _userManagementService = userManagementService;
+        }
+
+        public async Task Add(PaymentHistoryDto paymentDto)
+        {
+            var currentBalance = (await _userManagementService.GetUserById(paymentDto.UserId)).Details.Balance;
+
+            var mapper = new MapperConfiguration(cfg => cfg.CreateMap<PaymentHistoryDto, PaymentHistory>()
+                .ForMember(p => p.CurrentBalance, opt => opt.MapFrom(n => currentBalance)))
```

Comment:
What about add User navigation property to your PaymentHistoryDto and get balance directly in mapper without getting full user through user service?
And what about using ProjectTo here instead of Map?